examples/shp/gather_test.cpp

```diff
@@ -0,0 +1,128 @@
+// SPDX-FileCopyrightText: Intel Corporation
```

Comment:
Please add compiling of new binaries to cmake. They don't need to be executed, but let's keep CI checking its compilation at least.